__tests__/atomic-swap/sign-and-push.test.js

```diff
@@ -0,0 +1,127 @@
+import hathorLib from '@hathor/wallet-lib';
+import { ProposalInput, ProposalOutput, PartialTx } from '@hathor/wallet-lib/lib/models/partial_tx';
+import TestUtils from '../test-utils';
+
+const walletId = 'stub_atomic_swap_tx_proposal_sign_and_push';
+
+describe('tx-proposal sign-and-push api', () => {
+  const testnet = new hathorLib.Network('testnet');
+  const fakeTxId = '00003392e185c6e72d7d8073ef94649023777fd23c828514f505a7955abf0caf';
+  const fakeUid = '0000219a831aaa7b011973981a286142b3002cd04763002e23ba6fec7dadda44';
+  const createProposal = (inputs, outputs) => {
+    const partialTx = new PartialTx(testnet);
+    partialTx.inputs = inputs;
+    partialTx.outputs = outputs;
+
+    const proposal = new hathorLib.PartialTxProposal(testnet);
+    proposal.partialTx = partialTx;
+    return proposal;
+  };
+  const scriptFromAddress = base58 => {
+    const p2pkh = new hathorLib.P2PKH(new hathorLib.Address(base58, { network: testnet }));
+    return p2pkh.createScript();
+  };
+
+  const spy = jest.spyOn(hathorLib.PartialTxProposal, 'fromPartialTx');
+  const spyValidate = jest.spyOn(PartialTx.prototype, 'validate')
+    .mockImplementation(async () => true);
+
+  beforeAll(async () => {
+    await TestUtils.startWallet({ walletId, preCalculatedAddresses: TestUtils.addresses });
+  });
+
+  afterAll(async () => {
+    await TestUtils.stopWallet({ walletId });
+    // cleanup mock
+    spy.mockRestore();
+    spyValidate.mockRestore();
+  });
+
+  afterEach(() => {
+    spy.mockClear();
+  });
+
+  it('should fail if partial_tx is not a string', async () => {
```

Comment:
What if we don't send the required parameters?

Reply:
Added all invalid params cases to the tests.